### PR TITLE
Authentication 🗝 - Sandra Shumanteva and Sofia Vaz Sousa

### PR DESCRIPTION
**Pull request by sofiavazs**
Link to my deployed project:
Frontend : https://auth-sandra-sofia.netlify.app/ 
Backend: https://authentication-sandra-sofia.herokuapp.com/

My reflections on how this week's project turned out:
Happy that we manage to meet the blue level requirements. It was a really tough week, as I worked a lot on my normal job.

### CODE REVIEW

File: frontend/src/components/SignIn.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        }
+        return response.json();
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFailed(err));
+  };
+
+  const handleLogout = (loggingout) => { 
+    dispatch(user.actions.logout({ logout: loggingout }));
+    window.location.reload()
```
Comment (IngelaL) [inline frontend/src/components/SignIn.js:69]: Great that you used window.location.reload. Maybe add ; after ( ).

File: frontend/src/components/SignIn.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  };
+
+  if(accessToken) {
+    return <>
+    <h2>{`Hello ${name} you are logged in! shhh this is a secret area just for you`}</h2>
```
Comment (IngelaL) [inline frontend/src/components/SignIn.js:74]: Nice with a personal message to the user.

File: frontend/src/components/SignIn.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  return (
+    <section className="login-form">
+      <form className="form">
+        <h1>Sign Up/Login</h1>
+        <label>
+          Name
+          <input
+            required
+            minLength="5"
```
Comment (IngelaL) [inline frontend/src/components/SignIn.js:89]: You are using minlength 3 in backend so you need to change this one in frontend to 3.

File: frontend/src/components/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            value={name}
+            onChange={(event) => setName(event.target.value)}
+          />
+        </label>
+        <label>
+          Password
+          <input
+            type="password"
+            required
+            minLength="5"
```
Comment (IngelaL) [inline frontend/src/components/SignIn.js:99]: This is not working so you could delete it or add minlength with a schema in backend.

**Review by IngelaL — CHANGES_REQUESTED** [review]
Hi Sandra and Sofia! Your page looks really nice and the code is well structured with good indentation. The error handling seems to work as it should. However, there are a few things you should fix: It is possible to sign up and log in without a password. Also, the secret message should be stored in backend as we have understood it. Overall, great job 👏 //Ingela and Jessica

File: backend/server.js [hunk 5]
```diff
@@ -1,29 +1,95 @@
...
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// User Model 
+const User = mongoose.model('User', {
+  name: {
+    type: String,
+    minlength:3,
+    unique: true,
+    required: true,
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply backend/server.js]: Fixed. It had to do with hashing the password without validating it first. So no validation was being applied 😊

File: backend/server.js [hunk 6]
```diff
@@ -1,29 +1,95 @@
...
+    res.status(201).json({id: user._id, accessToken: user.accessToken});
+  } catch (err) {
+    res.status(400).json({message: "Could not create user", errors:err.errors})
+  };
+});
+
+// SECURE ENDPOINT
+app.get('/users/:id', authenticateUser);
+app.get('/users/:id', (req, res) => {
+  res.status(401).send();
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply backend/server.js]: Fixed 😊

File: frontend/src/components/SignIn.js (hunk 1 above)
Comment (sofiavazs) [reply frontend/src/components/SignIn.js]: Done 😊

**Review by IngelaL — APPROVED** [review]
Hi! Looks great now 🙌 One thing, in the comment to this PR you have the link to the heroku page also for frontend, so don't forget to change that to the netlify link!

**Review by KarolCislowski — APPROVED** [review]
Your friends gave you awesome feedback and it is great to see that you responded to it! This is very important to do ❤️ 

Your code is clean and well structured. I can see that you thought a lot before the execution.  🌈 

WELL DONE 🦄

**Comment by sofiavazs** [comment]
Thank you Karol! 🚀

**Pull request closed without merging**